fastapi/dependencies/utils.py

```diff
@@ -462,13 +462,10 @@ async def solve_dependencies(
 ]:
     values: Dict[str, Any] = {}
     errors: List[ErrorWrapper] = []
-    response = response or Response(
-        content=None,
-        status_code=None,  # type: ignore
-        headers=None,  # type: ignore # in Starlette
-        media_type=None,  # type: ignore # in Starlette
-        background=None,  # type: ignore # in Starlette
-    )
+    if response is None:
+        response = Response()
+        del response.headers["content-length"]
+        response.status_code = None
```

Comment:
@tomchristie FastAPI uses **a** `Response` object that travels through the dependencies, but that `Response` is not the response that will be returned at the end. There's a "cleanup" after the dependencies are solved. You can see the "cleanup" logic below:

https://github.com/tiangolo/fastapi/blob/291180bf2d8c39e84860c2426b1d58b6c80f6fef/fastapi/routing.py#L216-L254

I've selected the whole snippet, from the point we get the `Response` object used in the dependencies (`solve_dependencies` call) until the last 5 lines (on which the values from `Response` are passed to the actual response object that will be used.

This comment is JFYK, I don't expect any action to be taken on Starlette. Also, considering that FastAPI tweaks Starlette internally for some features it has.